### PR TITLE
Allow blockchain network type to be set from config.

### PR DESCRIPTION
**Pull request by ke6jjj**
Add the ability to configure the `libp2p` network to something other than `mainnet`,
if overridden by `sys.config`.

These changes should allow a node to follow the validators testnet (or any other
future net, for that matter).

### CODE REVIEW

File: config/val.config [hunk 1]
```diff
@@ -0,0 +1,40 @@
+%% -*- erlang -*-
+[
+ "config/sys.config",
```
Comment (madninja) [inline config/val.config]: This should really just be a testnet.config and override for sys.config not completely replacing all the below. That is, if we actually ever want to support testnet in the repo going forward. 

I don't think we should but at least the network setting in the sup and a bumped libp2p and core to pick up the crypto changes in v1.3.3 would be great

File: src/bn_sup.erl [hunk 2]
```diff
@@ -33,6 +33,8 @@ start_link() ->
 init([]) ->
     erlang:system_flag(fullsweep_after, 0),
 
+    ok = libp2p_crypto:set_network(application:get_env(blockchain, network, mainnet)),
```
Comment (madninja) [inline src/bn_sup.erl:36]: This is really the only fix we want beyond an upgraded core once that lands in master

**Comment by madninja** [comment]
Just missing setting the network type on startup?

**Comment by ke6jjj** [comment]
I'm not sure what that means? It has to be set in the config somewhere, but we don't want to check in a config in this PR, correct? That means this is the only change in the PR.

**Comment by madninja** [comment]
Yeah I missed the patch bit where crypto:set_network is called. all good

**Review by madninja — CHANGES_REQUESTED** [review]
While you updated the core tag the git reference in the lockfile appears _not_ to be pointing at that tagged commit on master

**Review by madninja — APPROVED** [review]

**Pull request merged**